src/transformers/models/umt5/modeling_umt5.py

```diff
@@ -240,41 +248,71 @@ def compute_bias(self, query_length, key_length, device=None):
         values = values.permute([2, 0, 1]).unsqueeze(0)  # shape (1, num_heads, query_length, key_length)
         return values
 
+    def _prepare_key_values(
```

Comment:
This abstraction does not look particularly useful here. However, for models with multiple attention implementations, this abstraction is useful: all attention implementations can share it! 

(e.g. in Bart the benefits are clear)